apis/config/v1beta1/defaults.go

```diff
@@ -73,10 +67,21 @@ func SetDefaults_Configuration(cfg *Configuration) {
 	if len(cfg.Health.HealthProbeBindAddress) == 0 {
 		cfg.Health.HealthProbeBindAddress = DefaultHealthProbeBindAddress
 	}
-	if cfg.LeaderElection != nil && cfg.LeaderElection.LeaderElect != nil &&
-		*cfg.LeaderElection.LeaderElect && len(cfg.LeaderElection.ResourceName) == 0 {
+
+	if cfg.LeaderElection == nil {
+		cfg.LeaderElection = &configv1alpha1.LeaderElectionConfiguration{}
+	}
+	if len(cfg.LeaderElection.ResourceName) == 0 {
 		cfg.LeaderElection.ResourceName = DefaultLeaderElectionID
 	}
+	if len(cfg.LeaderElection.ResourceLock) == 0 {
+		// Default to Lease as component-base still defaults to endpoint resources
+		// until core components migrate to using Leases. See k/k #80289 for more details.
+		cfg.LeaderElection.ResourceLock = resourcelock.LeasesResourceLock
+	}
+	// Use the default LeaderElectionConfiguration options
+	configv1alpha1.RecommendedDefaultLeaderElectionConfiguration(cfg.LeaderElection)
```

Comment:
As before, question about user-facing consequences of setting the defaults.

Reply:
With the values already set in the default Kueue configuration, i.e.: 

https://github.com/kubernetes-sigs/kueue/blob/d4d3f3be091eea414ae8d7018108ff9a391dc14d/config/components/manager/controller_manager_config.yaml#L10-L12

That PR is "idem-potent".

The only change would be if an existing user would use a configuration with `.leaderElection` or `.leaderElection.leaderElect` not defined, then the new default would be true, instead of false.

But I'd be inclined to think enabling leader election by default is desirable.

Reply:
Yes, the latter, their values are equal to the implicit defaults used before.